client/hostvolumemanager/host_volume_plugin.go

```diff
@@ -71,6 +77,14 @@ type HostVolumePluginDeleteResponse struct {
 const HostVolumePluginMkdirID = "mkdir"
 const HostVolumePluginMkdirVersion = "0.0.1"
 
+// HostVolumePluginMkdirParams represents the parameters{} that the "mkdir"
+// plugin will accept.
+type HostVolumePluginMkdirParams struct {
+	Uid  int         `mapstructure:"uid"`
+	Gid  int         `mapstructure:"gid"`
+	Mode os.FileMode `mapstructure:"mode"`
```

Comment:
The other configuration options are using JSON decoding and tags. Am I right in thinking we are using mapstructure here to perform direct decoding of `Mode` along with some potential other type/default help?

The `github.com/mitchellh/mapstructure` repo has been archived, so I just want to check we are importing it for reasons we cannot avoid and potentially do with JSON decoding and also keeping consistency in the file.

Reply:
the main things I wanted from mapstructure were:

1. automatic type conversion
1. error on unknown fields
1. detection of unset fields (not just zero values, because unset uid/gid should become -1)

I can do all these things manually, and I agree with @tgross [down yonder](https://github.com/hashicorp/nomad/pull/25533#discussion_r2016560365), too, so I'll refactor.

this is how I trick myself into doing TDD - write it, then test it, then re-write it from the tests :stuck_out_tongue_closed_eyes: